drafts/intro-to-r.md

```diff
@@ -1,151 +1,245 @@
-	      count = count + 1
-	    }
-	  }
-	  return((count*4)/trials)
-	}
-	
-	montecarloPi(1000)
+	print("Hello World!")
 
 R normally runs as an interactive shell, but it is easy to run in batch mode too.
 
-	$ Rscript --no-save mcpi.R
-	[1] 3.141956
+	$ Rscript --no-save hello_world.R
+	[1] "Hello World!"
 
-This should take few seconds to run. Now edit the file. Increasing the trials ten times (10000000) it will take little over a minute to run, but the estimation still isn't very good. Fortunately, this problem is pleasingly parallel since we're just sampling random points. So what do we need to do to run R on the campus grid?
+Notice here that we're using Rscript (equivalent to `R CMD BATCH`) which accepts the script as command line argument. This approach makes `R` much less verbose, and it's easier to parse the output later. If you run it at the command line, you should get similar output as above.
 
-##Building the HTCondor job
-The first thing we're going to need to do is create a wrapper for our R environment, based on the setup we did in previous sections. Create the file `R-wrapper.sh`:
+### Build the HTCondor job
+
+To prepare our R job to run on OSG, we need to create a wrapper for our R environment, based on the setup we did in previous sections. Create the file `R-wrapper.sh`:
 
 	#!/bin/bash
 	 
-	EXPECTED_ARGS=1
-	 
-	if [ $# -ne $EXPECTED_ARGS ]; then
-	  echo "Usage: R-wrapper.sh file.R"
-	  exit 1
-	else
-	  source /cvmfs/oasis.opensciencegrid.org/osg/modules/lmod/current/init/bash
-	  module load R
-	  module load libgfortran
-	  Rscript $1
-	fi
+    module load r/3.5.1-py2.7
+    Rscript --no-save hello_world.R
 
-Notice here that we're using Rscript (equivalent to `R --slave`). It accepts the script as command line argument, it makes `R` much less verbose, and it's easier to parse the output later. If you run it at the command line, you should get similar output as above. This lets the wrapper launch `R` on any generic worker node under HTCondor.
+Change the permissions on the wrapper script so it is executable and then test it for correct output:
 
-	$ ./R-wrapper.sh mcpi.R
-	[1] 3.142524
+	$ chmod +x R_wrapper.sh
+	$ ./R-wrapper.sh
+	[1] "Hello World!"
 
 Now that we've created a wrapper, let's build a HTCondor submit file around it. We'll call this one `R.submit`:
 
 	universe = vanilla
-	log = mcpi.log.$(Cluster).$(Process)
-	error = mcpi.err.$(Cluster).$(Process)
-	output = mcpi.out.$(Cluster).$(Process)
-		 
-	# Setup R path, run the mcpi.R script
+	log = R.log.$(Cluster).$(Process)
+	error = R.err.$(Cluster).$(Process)
+	output = R.out.$(Cluster).$(Process)
+	 
 	executable = R-wrapper.sh
-	transfer_input_files = mcpi.R
-	arguments = mcpi.R
-		 
-	requirements = OSGVO_OS_STRING == "RHEL 6" && Arch == "X86_64" && HAS_MODULES == True
-	queue 10 
+	transfer_input_files = hello_world.R
+	 
+	requirements = OSGVO_OS_STRING == "RHEL 7" && Arch == "X86_64" && HAS_MODULES == True
+	queue 10
```

Comment:
Do we need to run 10 jobs? For test purposes, it feels like 1 would do...

Reply:
Agreed, this was an artifact of using the mpi example as a base - Changed this to 1.